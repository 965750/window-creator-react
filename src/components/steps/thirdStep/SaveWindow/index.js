import { connect } from 'react-redux'
import { FormattedMessage } from 'react-intl'
import { saveWindow } from '../../../../store/actions/creatorActions'

import BaseInput from '../../../inputs/BaseInput'
import BaseSubmit from '../../../inputs/BaseSubmit'
import PropTypes from 'prop-types'
import React, { Component } from 'react'

class SaveWindow extends Component {
  state = {
    name: '',
    error: '',
  }

  validationError = (error) => {
    this.setState({
      error,
    })
  }

  handleChange = (e) => {
    this.setState({
      [e.target.id]: e.target.value,
    })
  }

  saveProject = (e) => {
    e.preventDefault()

    this.props.saveWindow(this.state.name)
  }

  render() {
    return (
      <div>
        <p className="text-xs mb-2">
          <FormattedMessage
            id="You can save Your current window project for later use."
            defaultMessage="You can save Your current window project for later use."
          />
        </p>
        <form onSubmit={this.saveProject} className="flex mb-2">
          <BaseInput
            value={this.state.name}
            validationError={this.validationError}
            handleChange={this.handleChange}
            id="name"
            type="text"
            toValidate={['isRequired', 'tooLong']}
            classes="mr-1 w-full"
            placeholder="Name of Your project"
          />
          <BaseSubmit
            disabled={this.state.error || !this.state.name}
            sm
            value="Save project"
          />
        </form>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  saveWindow: (name) => dispatch(saveWindow(name)),
})

SaveWindow.propTypes = {
  saveWindow: PropTypes.func,
}

export default connect(null, mapDispatchToProps)(SaveWindow);
