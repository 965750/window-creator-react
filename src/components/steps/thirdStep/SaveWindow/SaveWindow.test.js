import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react'

import SaveWindow from './index'
import messages from '../../../../messages-i18n'
import thunk from 'redux-thunk'
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('SaveWindow', () => {
  let wrapper
  const initialProps = {
    saveWindow: jest.fn()
  }

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <SaveWindow {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
