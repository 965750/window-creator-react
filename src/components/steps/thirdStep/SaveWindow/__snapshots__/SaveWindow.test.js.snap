// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveWindow should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <p
      class="text-xs mb-2"
    >
      Możesz zapisać swój aktualny projekt okna do późniejszego wykorzystania.
    </p>
    <form
      class="flex mb-2"
    >
      <div
        class="field__wrapper mr-1 w-full"
        data-testid="inputWrapper"
      >
        <input
          class="field  border border-silver rounded p-3 w-full"
          id="name"
          placeholder="Nazwa Twojego projektu"
          type="text"
          value=""
        />
      </div>
      <input
        class="submit submit--sm  cursor-pointer block bg-mountainMeadow rounded text-white"
        data-testid="submitWrapper"
        disabled=""
        type="submit"
        value="Zapisz projekt"
      />
    </form>
  </div>
</DocumentFragment>
`;
